components/carousel/private/carousel-indicators.jsx

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// ### React
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// ### classNames
+// [github.com/JedWatson/classnames](https://github.com/JedWatson/classnames)
+// This project uses `classnames`, "a simple javascript utility for conditionally
+// joining classNames together."
+import classnames from 'classnames';
+
+import { CAROUSEL_INDICATORS } from '../../../utilities/constants';
+
+/**
+ * CarouselIndicators is used to display the list of indicators associated to the number of panels
+ * a carousel has
+ */
+const CarouselIndicators = (props) => (
+	<ul
+		className="slds-carousel__indicators slds-col slds-text-align_center"
+		role="tablist"
+	>
+		{[...Array(props.noOfIndicators).keys()].map((key) => {
+			const index = key + 1;
+			const isSelectedPanel = index === props.currentIndex;
+			const indicatorActionClassName = classnames(
+				'slds-carousel__indicator-action',
+				props.className,
+				{
+					'slds-is-active': isSelectedPanel,
+				}
+			);
+			return (
+				<li
+					className="slds-carousel__indicator slds-m-horizontal_xx-small"
+					role="presentation"
+					key={index}
+				>
+					<a
```

Comment:
In order to move the focus to the active dot link. You will want to pass down from internal state an `indicatorHasFocus` and `isSelectedPanel` to the active dot. Then when those are true set the ref callback to call `focus()` on the node.

Reply:
done